ballerina-to-openapi/src/main/java/io/ballerina/openapi/service/mapper/example/parameter/ParameterExampleMapper.java

```diff
@@ -0,0 +1,102 @@
+/*
+ *  Copyright (c) 2024, WSO2 LLC. (http://www.wso2.org).
+ *
+ *  WSO2 LLC. licenses this file to you under the Apache License,
+ *  Version 2.0 (the "License"); you may not use this file except
+ *  in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing,
+ *  software distributed under the License is distributed on an
+ *  "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ *  KIND, either express or implied.  See the License for the
+ *  specific language governing permissions and limitations
+ *  under the License.
+ */
+package io.ballerina.openapi.service.mapper.example.parameter;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import io.ballerina.compiler.api.SemanticModel;
+import io.ballerina.compiler.api.symbols.AnnotationAttachmentSymbol;
+import io.ballerina.compiler.api.symbols.TypeSymbol;
+import io.ballerina.openapi.service.mapper.diagnostic.DiagnosticMessages;
+import io.ballerina.openapi.service.mapper.diagnostic.ExceptionDiagnostic;
+import io.ballerina.openapi.service.mapper.diagnostic.OpenAPIMapperDiagnostic;
+import io.ballerina.openapi.service.mapper.example.ExamplesMapper;
+import io.ballerina.tools.diagnostics.Location;
+import io.swagger.v3.oas.models.examples.Example;
+import io.swagger.v3.oas.models.media.Schema;
+import io.swagger.v3.oas.models.parameters.Parameter;
+
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+
+import static io.ballerina.openapi.service.mapper.Constants.EXAMPLE;
+import static io.ballerina.openapi.service.mapper.Constants.EXAMPLES;
+import static io.ballerina.openapi.service.mapper.example.CommonUtils.setExampleForInlineRecordFields;
+
+/**
+ * This {@link ParameterExampleMapper} class represents the examples mapper for resource function parameter.
+ *
+ * @since 2.1.0
+ */
+public abstract class ParameterExampleMapper extends ExamplesMapper {
+
+    Parameter parameterSchema;
+    List<AnnotationAttachmentSymbol> annotations;
+    List<OpenAPIMapperDiagnostic> diagnostics;
+    String paramName;
+    Location location;
+    String paramTypeName;
+    TypeSymbol paramType;
+
+    protected ParameterExampleMapper(List<AnnotationAttachmentSymbol> annotations, Parameter parameterSchema,
```

Comment:
Shall we have one data store model for parsing all the constructor parameters, since this contains more than 5?

Reply:
This is a abstract class constructor which is intended to be used inside the package. Will make it package-private

Reply:
The constructor is made as package-private with [2764e2d](https://github.com/ballerina-platform/openapi-tools/pull/1740/commits/2764e2dd6bee1a94d6c8980d3532e59d8019f545)